Assets/PatchKit Patcher/Scripts/AppData/Remote/RemoteResourceDownloader.cs

```diff
@@ -126,6 +126,31 @@ private bool AreChunksAvailable()
         }
 
         public void Download(CancellationToken cancellationToken)
+        {
+            int retriesLeft = RetriesCount;
+            do
+            {
+                try
+                {
+                    ResolveDownloader(cancellationToken);
+                }
+                catch (Exception ex)
+                {
+                    if (retriesLeft > 0)
+                    {
+                        retriesLeft--;
+                        DebugLogger.LogWarningFormat("Resolving Dowloader failed, retry: {0}/{1}", RetriesCount-retriesLeft+1, RetriesCount);
+                    }
+                    else
+                    {
+                        DebugLogger.LogErrorFormat("Resolving Dowloader failed, no retries left, throwing further");
+                        throw ex;
+                    }
+                }
+            } while (retriesLeft > 0);
+        }
+
+        private void ResolveDownloader(CancellationToken cancellationToken)
         {
             Assert.MethodCalledOnlyOnce(ref _downloadHasBeenCalled, "Download");
```

Comment:
This assertion must be moved to `Download` - `ResolveDownloader` is called multiple times by `Download` so it would throw `AssertionException`.